video/src/ai/eto/rikai/sql/spark/datasources/VideoOptions.scala

```diff
@@ -1,21 +1,65 @@
 package ai.eto.rikai.sql.spark.datasources
 
 import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.bytedeco.ffmpeg.global.swscale
+import org.bytedeco.javacv.FrameGrabber
 
 class VideoOptions(@transient val parameters: CaseInsensitiveMap[String])
     extends Serializable {
   def this(parameters: Map[String, String]) = {
     this(CaseInsensitiveMap(parameters))
   }
 
+  // Number of frames per second
   val fps = parameters.get("fps").map(_.toInt).getOrElse(1)
+  // Frame step size and the frame_id to start
+  val frameStepSize = parameters.get("frameStepSize").map(_.toInt).getOrElse(0)
+  val frameStartId = parameters.get("frameStartId").map(_.toInt).getOrElse(0)
 
   val imageWidth = parameters.get("imageWidth").map(_.toInt).getOrElse(0)
-
   val imageHeight = parameters.get("imageHeight").map(_.toInt).getOrElse(0)
 
   /** A list of scaler flags from
     * http://www.ffmpeg.org/ffmpeg-scaler.html#toc-Scaler-Options
     */
   val scalerFlag = parameters.get("scalerFlag").getOrElse("bilinear")
+
+  def getFrameStep(grabber: FrameGrabber): (Int, Int) = {
+    if (frameStepSize > 0) {
+      val startId = if (frameStartId == 0) frameStepSize else frameStartId
```

Comment:
Why user will not need a frame start id `0`?

Reply:
Because frame_id starts from 1

Reply:
It should be documented.